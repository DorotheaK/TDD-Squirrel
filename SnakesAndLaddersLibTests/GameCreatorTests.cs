using FakeItEasy;
using FluentAssertions;
using FluentAssertions.Execution;
using SnakesAndLaddersLib;
namespace SnakesAndLaddersLibTests;

    public class GameCreatorTests
    {


        [TestCase(10)]
        [TestCase(15)]
        public void CreateGame_Should_Return_InitializedGame_ForGivenNumberOfFields(int numberOfFields)
        {
            var game = GameCreator.CreateGame(numberOfFields);

            using var scope = new AssertionScope();
            game.Should().NotBeNull();
            game.IsDieDisabled.Should().BeFalse();
            game.Status.Should().BeTrue();
            game.Position.Should().Be(0);
            game.NumberOfFields.Should().Be(numberOfFields);
        }

        [Test]
        public void CreateGame_Should_Return_SquareGame()
        {
            var game = GameCreator.CreateGame(1);

            using var scope = new AssertionScope();
            game.Should().NotBeNull();
            game.Status.Should().BeTrue();
            game.Position.Should().Be(0);
            game.NumberOfFields.Should().Be(1);
            game.Rows.Should().Be(1);
            game.Columns.Should().Be(1);
    }
    }
    
