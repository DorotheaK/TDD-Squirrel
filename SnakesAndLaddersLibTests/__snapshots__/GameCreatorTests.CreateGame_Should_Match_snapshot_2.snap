{
  "IsDieDisabled": false,
  "Status": true,
  "Rows": 2,
  "Columns": 2
}
